### ISSUE TITLE
NewFileAppendTransaction() - AppendTx : exceptional precheck status DUPLICATE_TRANSACTION received

### ISSUE THREAD

**Issue by tomzhenghedera** [hashgraph/hedera-sdk-go#489]
### Description

For `v2.14.0-beta.1`(other later releases has compile problems reported in [Issue 488](https://github.com/hashgraph/hedera-sdk-go/issues/488)) `NewFileAppendTransaction()` failed 

```
03:33:18.491 INFO get_file[UpdateState.bin]: contract bytecode size: [6572] bytes
03:33:21.764 ERROR get_file: AppendTx : exceptional precheck status DUPLICATE_TRANSACTION received for transaction 0.0.67560@1651653189.47393816
```
The same program was able to load the same BIN file at `v2.11.0-beta.1` as well as file at double the size.  

### Steps to reproduce

The error occurred at `line 64`
```
 37     // Upload a file containing the byte code
 38     byteCodeTransactionResponse, err := hedera.NewFileCreateTransaction().
 39         SetNodeAccountIDs([]hedera.AccountID{node}).
 40         SetMaxTransactionFee(hedera.NewHbar(fee)).
 41         SetKeys(client.GetOperatorPublicKey()).
 42         SetContents([]byte("Dummy File")).
 43         Execute(client)
 44 
 45     if err != nil {
 46         log.Errorf("get_file: error create file tx: %v", err.Error())
 47         return hedera.FileID{}, err
 48     }
 49 
 50     byteCodeTransactionReceipt, err := byteCodeTransactionResponse.GetReceipt(client)
 51     if err != nil {
 52         log.Errorf("get_file: GetReceipt: %v", err.Error())
 53         return hedera.FileID{}, err
 54     }
 55 
 56     byteCodeFileID := *byteCodeTransactionReceipt.FileID
 57 
 58     appendResponse, err := hedera.NewFileAppendTransaction().
 59         SetFileID(byteCodeFileID).
 60         SetContents([]byte(smartContractByteCode)).
 61         SetMaxTransactionFee(hedera.NewHbar(fee)).
 62         Execute(client)
 63     if err != nil {
 64         log.Errorf("get_file: AppendTx : %v", err.Error())
 65         return hedera.FileID{}, err
 66     }
 67     _, err = appendResponse.GetReceipt(client)
 68     if err != nil {
 69         log.Errorf("get_file: AppendTx GetReceipt: %v", err.Error())
 70         return hedera.FileID{}, err
 71     }
```

### Hedera network

previewnet

### Version

v2.14.0-beta.1

### Operating system

Linux

**Comment by andrix10** [comment]
I am not sure, couldn't reproduce it even when I use very large array for contents, on both previewnet or testnet, with or without NodeAccountIDs set.

**Comment by tomzhenghedera** [comment]
I re-compiled the code with v2.13.2 and tried on previewnet and got the same, this time assumes the code is compiled correctly.  

The bytecode is loaded from a file but should be no difference vs a pre-allocated byte array. What is curios is the error type

```
AppendTx : exceptional precheck status DUPLICATE_TRANSACTION received for transaction 0.0.67560@1651692667.891400634
```
this is just one transaction being submitted and how could it got `DUPLICATE_TRANSACTION` escapes me.

**Comment by andrix10** [comment]
Ye I have no idea, had a bug with that before v2.13.2 but it was when 14 transactions were being sent, that's now fixed and also made mocking tests which show everything to be working so far.

**Comment by tomzhenghedera** [comment]
I did some digging today and  the failures I'd seen, for example,
```
23:53:38.805 INFO get_file[call.bin]: contract bytecode size: [2646] bytes
23:53:51.876 ERROR get_file: AppendTx : exceptional precheck status DUPLICATE_TRANSACTION received for transaction 0.0.1017@1651812813.811942026
```
may be related to 
```
    the size of the bytecode vs SetMaxChunkSize(<value>).
```
Using above `call.bin` file as example, it has a size of `2646` bytes, 
- if calling `NewFileAppendTransaction()` without setting `SetMaxChunkSize()`, using the default(not sure what's the default chunk size), the code would fail with `DUPLICATE_TRANSACTION`

- if calling `NewFileAppendTransaction().SetMaxChunkSize(4096)`, the code would pass and a file was successfully created

- if calling `NewFileAppendTransaction().SetMaxChunkSize(2048)`, the code would fail again
 with `DUPLICATE_TRANSACTION`

I tried with some larger contracts that would have the same behavior where `SetMaxChunkSize()` had to be specified correctly based on the size of the bytecode.  Exception is if the file gets larger, where I have a BIN file at 14K, then if I provided `SetMaxChunkSize(15000)` it would give a different error
```
00:08:55.306 INFO get_file[ERC20.bin]: contract bytecode size: [14930] bytes
00:09:01.076 ERROR get_file: AppendTx : exceptional precheck status TRANSACTION_OVERSIZE received for transaction 0.0.1017@1651813729.792882206
```
at `v2.11.0-beta.1` or early releases the code would have no issue loading this same BIN file as it is one of the contracts I'd been using running smartcontract tests since last year.

I think something changed related to the `SetMaxChunkSize` in the latest releases for `FileAppend`.

**Comment by andrix10** [comment]
Ye I am not exactly sure I will ask @danielakhterov to run some tests too, I ran a bunch of stuff today, 2646 with max chunks 4096 and 2048, those worked. With and without node set, no max chunks set, unit and e2e tests. Even printed out raw transactions it all seemed to work fine. O then also same for mocking tests. 

Only time it error with `TRANSACTION_OVERSIZE` is when i had max chunks at 15000 and my file was 10000

**Comment by tomzhenghedera** [comment]
Thanks! My client didn't see `TRANSACTION_OVERSIZE` while loading the same file with the `14930` size and the `SetMaxChunkSize ` was originally set at `4096` and worked with both small and large sized files.  The behavior changed while moving to later releases(traced back to `v2.13.0`. `v2.13.1` saw same errors)

I will try with the `v2.14.0-beta.2` today, not sure will make any diff. as I didn't much changes with `v2.14.0-beta.1`

**Comment by tomzhenghedera** [comment]
`v2.14.0-beta.2` shows the same, in this example, a file size just below `7000`

with `SetMaxChunkSize(7000)`
```
12:37:17.940 INFO get_file[UpdateState.bin]: contract bytecode size: [6572] bytes
12:37:19.614 ERROR get_file: AppendTx : exceptional precheck status TRANSACTION_OVERSIZE received for transaction 0.0.3280@1652204228.353445227
```

with `SetMaxChunkSize(4096)`
```
12:35:35.075 INFO get_file[UpdateState.bin]: contract bytecode size: [6572] bytes
12:35:38.453 ERROR get_file: AppendTx : exceptional precheck status DUPLICATE_TRANSACTION received for transaction 0.0.3280@1652204125.564385547
```

Here is the go.mod file used, i am on `go 1.17`, I believe the sdk is on `go 1.13`, but my previous client was already on `1.17` so shouldn't matter.

```
go 1.17

require (
        github.com/davecgh/go-spew v1.1.1
        github.com/hashgraph/hedera-sdk-go/v2 v2.14.0-beta.2
        github.com/prometheus/client_golang v1.0.0
)

require (
        github.com/andres-erbsen/clock v0.0.0-20160526145045-9e14626cd129 // indirect
        github.com/beorn7/perks v1.0.0 // indirect
        github.com/btcsuite/btcd/btcec/v2 v2.1.2 // indirect
        github.com/decred/dcrd/dcrec/secp256k1/v4 v4.0.1 // indirect
        github.com/ethereum/go-ethereum v1.10.17 // indirect
        github.com/golang/protobuf v1.5.2 // indirect
        github.com/hashgraph/hedera-protobufs-go v0.2.1-0.20220426085224-6f0ec6b1f440 // indirect
        github.com/matttproud/golang_protobuf_extensions v1.0.1 // indirect
        github.com/pkg/errors v0.9.1 // indirect
        github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4 // indirect
        github.com/prometheus/common v0.6.0 // indirect
        github.com/prometheus/procfs v0.0.2 // indirect
        github.com/rs/zerolog v1.26.1 // indirect
        github.com/tyler-smith/go-bip39 v1.1.0 // indirect
        github.com/youmark/pkcs8 v0.0.0-20201027041543-1326539a0a0a // indirect
        golang.org/x/crypto v0.0.0-20211215165025-cf75a172585e // indirect
        golang.org/x/net v0.0.0-20210907225631-ff17edfbf26d // indirect
        golang.org/x/sys v0.0.0-20210906170528-6f6e22806c34 // indirect
        golang.org/x/text v0.3.7 // indirect
        google.golang.org/genproto v0.0.0-20200526211855-cb27e3aa2013 // indirect
        google.golang.org/grpc v1.45.0 // indirect
        google.golang.org/protobuf v1.26.1-0.20210525005349-febffdd88e85 // indirect
)
```

**Comment by SimiHunjan** [comment]
@tomzhenghedera can you post the code here to reproduce the issue?

**Comment by andrix10** [comment]
For the first one I also got `TRANSACTION_OVERSIZE` the second one passed perfectly though. I am also using go 1.16 myself. I did ask @danielakhterov to run some of these when he is free, to see if he can get similar errors on his machine.

**Comment by tomzhenghedera** [comment]
Here is a tar gz(tgz) that has my trimmed down code,

[file-append.gz](https://github.com/hashgraph/hedera-sdk-go/files/8676524/file-append.gz)

I believe I found a way to make the program to work with any sized bytecodes, what is needed is to add `nodeID` like the following

```go
    appendResponse, err := hedera.NewFileAppendTransaction().
           SetNodeAccountIDs([]hedera.AccountID{node}).
```

My early code didn't have this set, and failed while moving to the latest versions.  It's unclear to me why/how by adding the `nodeID` to make the call working again.  There is a `README.md` that has more details on how how I tested it.

Without `nodeID` set, the code will have to rely on `SetMaxChunkSize` value, but looks like it can only support upto just below `7000` bytes.

I think there is minimally a workaround available for my original problem, so this issue could be closed.  Though I think maybe worth to look into why the behavior of the call changed in the latest releases.
